File: 2958-length-of-longest-subarray-with-at-most-k-frequency/2958-length-of-longest-subarray-with-at-most-k-frequency.py
class Solution:
    def maxSubarrayLength(self,nums,k):
        n=len(nums)
        freq={}
        i=0
        j=0
        res=0

        while j<n:
            freq[nums[j]]=freq.get(nums[j],0)+1

            while freq[nums[j]]>k:
                freq[nums[i]]-=1
                i+=1

            res=max(res,j-i+1)
            j+=1

        return res

File: 2958-length-of-longest-subarray-with-at-most-k-frequency/2958-length-of-longest-subarray-with-at-most-k-frequency.cs
public class Solution {
    public int MaxSubarrayLength(int[] nums,int k) {
        int n=nums.Length;
        Dictionary<int,int>freq=new Dictionary<int,int>();

        int i=0;
        int j=0;
        int res=0;

        while(j<n){
            if(freq.ContainsKey(nums[j])){
                freq[nums[j]]++;
            }
            else{
                freq[nums[j]]=1;
            }

            while(freq[nums[j]]>k){
                freq[nums[i]]--;
                i++;
            }
            res=Math.Max(res,j-i+1);
            j++;
        }
        return res;
    }
}